Add support for Facebook Deep Linking + handle fb[appId] url schemes for Facebook Login

Morkos opened the pull request:
Hi there,

Please let me know your thoughts as without this commit, our users can't login with Facebook (when Facebook App is installed) when we use this branch library.

Also, I added support for deep linking to Facebook.

Best,
Mark

RodrigoSMarques commented:
Hi @Morkos 
Create a new PR to branch labelled "enhancement".
Before merge, i need more information about the problem.
This plugin not work with Facebook Login? Facebook URL Schemes not call? This is the problem?
If yes, I need run some tests and I believe know what is.
I will check the your code too.

Morkos commented:
Hi @RodrigoSMarques ,

Correct, it does **not** work with the Facebook Login. Facebook tries to call the URL scheme fb[appId]://authorize after login but branch does not handle this URL scheme correctly so the app does not get opened. I added a fix for this on Line 65 in SwiftFlutterBranchSdkPlugin.swift.

I added two other features as well.

To allow deeplinks from Facebook to work appropriately [1] and improve attribution for people using branch links on Apple Search Ads [2].

[1]: https://help.branch.io/using-branch/docs/facebook-app-install-ads (see registerFacebookDeepLinkingClass method)
[2]: https://help.branch.io/developers-hub/docs/ios-advanced-features#section-track-apple-search-ads

I will submit a new PR.

Best,
Mark
